client/material/menu/menu.tsx

```diff
@@ -53,175 +52,122 @@ export const Overlay = styled(CardLayer)<{ $dense?: boolean }>`
-      this.overlay.current!.scrollTop = itemHeight * newIndex
+      overlayRef.current.scrollTop = itemHeight * newIndex
     } else {
       // Make the new index the bottom item
-      this.overlay.current!.scrollTop = itemHeight * (newIndex + 1 - itemsShown)
+      overlayRef.current.scrollTop = itemHeight * (newIndex + 1 - itemsShown)
     }
-  }
-
-  onKeyDown = (event: KeyboardEvent) => {
-    if (event.code === ESCAPE) {
-      this.props.onDismiss()
-      return true
-    } else if (event.code === UP) {
-      this.moveActiveIndexBy(-1)
-      return true
-    } else if (event.code === DOWN) {
-      this.moveActiveIndexBy(1)
-      return true
-    } else if (event.code === TAB) {
-      if (this.state.activeIndex >= 0) {
-        this.onItemSelected(this.state.activeIndex)
+  })
+
+  useKeyListener({
+    // TODO(2Pac): Call the click handler of the currently active menu item by pressing Enter (and
+    // Space?)
+    onKeyDown: useStableCallback(event => {
+      if (event.code === UP) {
+        moveActiveIndexBy(-1)
         return true
-      }
-    } else if (event.code === ENTER || event.code === ENTER_NUMPAD) {
-      if (this.state.activeIndex >= 0) {
-        this.onItemSelected(this.state.activeIndex)
+      } else if (event.code === DOWN) {
+        moveActiveIndexBy(1)
         return true
+      } else if (event.code === ENTER || event.code === ENTER_NUMPAD || event.code === SPACE) {
+        const activeItem = menuItems[activeIndex]
+        if (isMenuItem(activeItem)) {
+          activeItem.props.onClick()
+        }
       }
-    }
 
-    return false
-  }
+      return false
+    }),
+  })
 
-  onItemSelected = (index: number) => {
-    if (this.props.onItemSelected) {
-      this.props.onItemSelected(index)
-    }
-  }
+  // We're using this variable as an index for only the menu items, so we don't try focusing a
+  // non-menu-item (e.g. overline, divider, etc.)
+  let i = 0
+  const items = React.Children.map(children, child => {
+    // Leave the non-selectable elements (e.g. dividers, overlines, etc.) as they are
+    if (!isMenuItem(child)) return child
+
+    const index = i
+    const elem = React.cloneElement(child, {
+      dense,
+      focused: index === activeIndex,
+    })
+    i++
+
+    return elem
+  })
+
+  return (
+    <Overlay key='menu' ref={overlayRef} className={className} $dense={dense}>
+      <Padding />
+      {items}
+      <Padding />
+    </Overlay>
+  )
+}
+
+export interface MenuProps extends MenuListProps {
+  popoverProps: Omit<PopoverProps, 'children'>
+}
+
+/**
+ * A wrapper component around Popover that can be used to quickly write menus.
+ *
+ * This component just renders the `Popover` and expects it to only render the `MenuList` when the
+ * menu is actually open. This makes sure that all the logic of the menu (e.g. keyboard handlers) is
+ * only run when the menu is open.
+ */
+export function Menu({ popoverProps, ...menuListProps }: MenuProps) {
```

Comment:
I really do not see the value in this existing, it just adds a component that can almost never be skipped during rendering (because its props contain an object whose reference will nearly always change), and doesn't do anything better than components just using these two components themselves